satpy/resample.py

```diff
@@ -275,8 +278,164 @@ def compute(self, data, weight_funcs=None, fill_value=None, with_uncert=False, *
+        lat_arr = lats.filled(np.nan).astype(np.float64)
+        fill_in = np.nan
+        grid_name = getattr(self.target_geo_def, "name", "N/A")
+        p = self.target_geo_def.proj4_string
+        cw = self.target_geo_def.pixel_size_x
+        ch = -abs(self.target_geo_def.pixel_size_y)
+        w = self.target_geo_def.x_size
+        h = self.target_geo_def.y_size
+        ox = self.target_geo_def.area_extent[0]
+        oy = self.target_geo_def.area_extent[3]
+
+        is_static = True  # SatPy/PyResample don't support dynamic grids out of the box yet
+        if is_static:
+            # we are remapping to a static unchanging grid/area with all of
+            # its parameters specified
+            # inplace operation so lon_arr and lat_arr are written to
+            swath_points_in_grid = _ll2cr.ll2cr_static(lon_arr, lat_arr, fill_in,
+                                                        p, cw, ch, w, h, ox, oy)
+        else:
+            results = _ll2cr.ll2cr_dynamic(lon_arr, lat_arr, fill_in, p, cw, ch,
+                                           width=w, height=h, origin_x=ox, origin_y=oy)
+            swath_points_in_grid, lon_orig, lat_orig, origin_x, origin_y, width, height = results
+
+            # edit the grid information because now we know what it is
+            # FIXME: This should be less magical to the user, maybe a separate step for this operation
+            self.target_geo_def.origin_x = origin_x
+            self.target_geo_def.origin_y = origin_y
+            self.target_geo_def.width = width
+            self.target_geo_def.height = height
+
+        # Determine if enough of the input swath was used
+        fraction_in = swath_points_in_grid / float(lon_arr.size)
+        swath_used = fraction_in > self.swath_usage
+        if not swath_used:
+            LOG.info("Data does not fit in grid %s because it only %f%% of the swath is used" % (grid_name, fraction_in * 100))
+            raise RuntimeError("Data does not fit in grid %s" % (grid_name,))
+        else:
+            LOG.debug("Data fits in grid %s and uses %f%% of the swath", grid_name, fraction_in * 100)
+
+        # it's important here not to modify the existing cache dictionary.
+        self.cache = {
+            "source_geo_def": source_geo_def,
+            "rows": lat_arr,
+            "cols": lon_arr,
+        }
+
+        self.caches[kd_hash] = self.cache
+        while len(self.caches) > CACHE_SIZE:
+            self.caches.popitem(False)
+
+        if cache_dir:
+            # XXX: Look in to doing memmap-able files instead `arr.tofile(filename)`
+            self.dump(filename)
+        return self.cache
+
+    def compute(self, data,
+                weight_count=10000, weight_min=0.01, weight_distance_max=1.0,
+                weight_sum_min=-1.0, maximum_weight_mode=False,
+                **kwargs):
+        rows = self.cache["rows"]
+        cols = self.cache["cols"]
+        out_arr = np.empty(data.shape, dtype=data.dtype)
```

Comment:
Actually I think this is a bug. I think it should be `rows.shape` instead of `data.shape` which is why I did `empty` in the first place.